demo/admin/src/dam/unsplash/ImportFromUnsplash.tsx

```diff
@@ -0,0 +1,77 @@
+import { useApolloClient } from "@apollo/client";
+import { CancelButton, messages, SaveButton } from "@comet/admin";
+import { useDamAcceptedMimeTypes } from "@comet/cms-admin";
+import { AdditionalToolbarItem } from "@comet/cms-admin/lib/dam/DamTable";
+import { useFileUpload } from "@comet/cms-admin/lib/dam/DataGrid/fileUpload/useFileUpload";
+import { clearDamItemCache } from "@comet/cms-admin/lib/dam/helpers/clearDamItemCache";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import { getRandomUnsplashImage, UnsplashImage } from "@src/dam/unsplash/getRandomUnsplashImage";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import UnsplashIcon from "./UnsplashIcon";
+
+export const ImportFromUnsplash: AdditionalToolbarItem = () => {
+    const client = useApolloClient();
+    const { allAcceptedMimeTypes } = useDamAcceptedMimeTypes();
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [unsplashImage, setUnsplashImage] = React.useState<UnsplashImage>({ file: { acceptedFiles: [], fileRejections: [] } });
+
+    const { uploadFiles } = useFileUpload({
+        acceptedMimetypes: allAcceptedMimeTypes,
+        onAfterUpload: () => {
+            client.reFetchObservableQueries();
+            clearDamItemCache(client.cache);
+        },
+    });
+
+    const handleOpenDialog = async () => {
+        const image = await getRandomUnsplashImage();
+        setUnsplashImage(image);
+        setIsOpen(true);
+    };
+
+    const handleCloseDialog = () => {
+        setIsOpen(false);
+    };
+
+    const handleSave = () => {
+        uploadFiles(unsplashImage.file, undefined, "unsplash", unsplashImage.url);
```

Comment:
We could change the args to an options object, then you won't need to set `folderId` undefined and it would be more clear what the arguments mean

Reply:
Yes that is a good idea but I think it would make more sense to include this in a separate PR since this affects multiple places. What is your opinion?

Reply:
It's only called three times, I think we may do this in this PR. Also, please change the function definition to return a Promise, as it is an async function:

<img width="584" alt="image" src="https://github.com/vivid-planet/comet/assets/48853629/a2115b63-7f05-438a-97b9-9617c2690792">